torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1595,6 +1578,17 @@ def _lazy_init(self) -> None:
         # pass gradient computation (though this may not be true)
         self.reshard_after_forward = False
         self._exec_order_data.init(self)
+        # For all sharding strategies communication is performed through `communication_hook`:
```

Comment:
I am not sure why this assert is in `_lazy_init()`. How can these attributes be `None`?

Reply:
Thanks for spotting this. This check was unnecessary here.